### DISCUSSION TITLE
New FIOS html remote

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#512] [Show and tell]
This has some features that require changes to the sling server code to work.  
1) I can use this one remote for all 6 of my configs.  It autoselects the based on the running stream.
2) I auto insert the names of the boxes into the drop down menu.
If anyone wants to use this, I can tell you what changes are required to the server.
[FIOS_remote.zip](https://github.com/GerryDazoo/Slinger/files/10523291/FIOS_remote.zip)

**Comment by socelillyhammer** [comment]
Is this for the new set-top box that uses Bluetooth?

On the new set-top box that uses Bluetooth remote, Gerrys code chrrent ir codes do not work.

But I was able to fix the ir for the fios one by using an old remote.

Of anyone can help with explaining how I can extract the IR Codes I can pass them over to build a fios one remote

**Reply by mdehrlich** [reply]
This has nothing to do with bluetooth.  Slingboxes only send IR codes.  If slingplayer worked with your fios box, then the box must handle IR.

**Comment by socelillyhammer** [comment]
It did.  But for some reason the coding is different.  I don't know why.
I've always had a problem with the new fios one boxes.

**Comment by as400jockey** [comment]
This looks very interesting.  What do I need to do to get this to work?  the remote loads up in Chrome but doesn't work.

I'm using VLC to get the video.  I have two slingboxes running - one is M1Basement the other is M1Kitchen.  My link for the basement box is

    http://mc1vm2:8080/slingbox/M1Basement/

And the current fios remote address is

    http://mc1vm2.8080/Remote/M1Basement/

I have slinger installed on the c: drive of vm MC1VM2 .

Thanks!!

**Comment by mdehrlich** [comment]
The remote must be installed as a file on the server in the same directory with the slinger app.  You point to the remote in the slinger config file.  I took the auto filling of the avail configs out of the remote.  Just put the configs in as options where it says INSERT HERE.

**Comment by as400jockey** [comment]
thanks for that...

I've been a developer since punch cards, but I'm not familiar with this type of code.  I put the file in the directory with the slinger server:

![image](https://user-images.githubusercontent.com/117923177/231541349-800eca5b-eb8d-4e4e-8385-4375d2bdb4dd.png)

I edited the code as follows:

![image](https://user-images.githubusercontent.com/117923177/231543762-49587ebb-3f48-440a-9963-479b2bbf044e.png)

I open c:\slinger\Fios_Remote.html in Chrome.  I see M1Basement on the list:

![image](https://user-images.githubusercontent.com/117923177/231543927-2081927f-bf1d-4542-823d-e73381f2be37.png)

I edited c:\slinger\config.ini:

![image](https://user-images.githubusercontent.com/117923177/231545215-0ac3928d-e71e-4771-9269-c545cc72f72a.png)

But click anything and I got File Could Not Be Accessed:

![image](https://user-images.githubusercontent.com/117923177/231545541-cf7c6f59-2189-4d03-b4a9-4e80ff2c3aa1.png)

I figured out it was looking for something on the c: drive in c:\Remote\ .  so I created that directory. then it appeared I needed Fios_remote.html .in that directory.  so I copied it there.  then it appeared to want a directory c:\remote\m1basement\ so I created that, but didn't put anything in it.

The web page stays up when I click on something now, but the dos box with the slinger server running isn't sending anything to the slingbox.

So I'm getting warmer... but I'm still missing something that is likely obvious to others...

**Comment by as400jockey** [comment]
![image](https://user-images.githubusercontent.com/117923177/231571188-41fbc39e-43b5-4d4c-a6c3-b8589aaaddc5.png)

I dawned on me that I was using the wrong url to access the remote.  But when I go to http://mc1vm2:8080/Remote/M1Basement I get the above error....

**Comment by as400jockey** [comment]
![image](https://user-images.githubusercontent.com/117923177/231571859-e747b147-76d1-43a6-9853-73ac1b3692b7.png)

**Reply by mdehrlich** [reply]
The problem may be you have an older version of slinger.  The faulty statement was changed in the non-windows version.  I don't know if was changed in the windows version?

**Reply by mdehrlich** [reply]
The problem has to do with how the % replacement operator works.

**Comment by as400jockey** [comment]
Thanks for that info.  I found the v4 stuff for Windows.  Audio and video are working for my M1Kitchen box,  but not for M1Basement.  Only getting audio there.  will reboot some stuff and check the configs.

Remote works great.  Awesome work!  The drop-down is misbehaving, but I'll figure that out.  For now, if I open http://mc1vm2:8080/Remote/M1Kitchen it works.  If I open http://mc1vm2:8080/Remote/M1Basement it works.  I just can't use the drop down to switch around.  When I do, Chrome tries to go to http://mc1vm2:8080/Slinger instead of http://mc1vm2:8080/Remote/M1Basement .

Will tinker and report back later.

Thanks again to all for such an awesome gadget allowing us to continue to use this hardware that Sling pulled out from under us.

**Comment by as400jockey** [comment]
Found my missing video issue.

Rebooted slingbox.  No joy.
Rebooted HDMI to Component converter.  No joy.
Rebooted HDMI splitter (takes 1 hdmi from Fios box splits to 2 - one for TV, one for HDMI to Component converter.  BINGO.

So the only (and very minor) issue is the drop down box in the remote. 

Time to get some work done :)

Will save this tinkering for another day.

**Comment by as400jockey** [comment]
Thanks for that.  Works great.

What is “Restart” supposed to do?  It’s trying to

                http://mc1vm2:8080/Remote/Restart

**Comment by as400jockey** [comment]
works great!!!!

**Comment by as400jockey** [comment]
Your remote is brilliant...

I was able to tweak the channel-specific buttons for my channel numbers.  The 'options' button was coming up with On Demand, which is preferable for me.  So I changed the button text.

The only button that isn't working is "DVR".  It's sending "73" but that goes nowhere.  How do I find out what the decodes are for a Fios P265 or similar remote?

**Reply by mdehrlich** [reply]
Weird.   Same remote that I use.  You can try other codes by putting ?code into the channel input.  Can't imagine why 073 doesn't work.

**Comment by mdehrlich** [comment]
I messed up and lost my code for restart.  Don't use it much, so I didn't notice.  
Here is the new remote with restart fixed.  Just change the config names in the options. (still don't know why DVR doesn't work for you.)
[FIOS_Remote.html.txt](https://github.com/GerryDazoo/Slinger/files/11237803/FIOS_Remote.html.txt)

**Comment by as400jockey** [comment]
2 things... "Restart" tries to go to http://mc1vm2:8080/remote/Restart
I get "Not Found".

Second thing - I tested codes 1 thru 100 using your ?nnn thing.  My "DVR" is on 65.  My 73 does nothing.  Very interesting.

Here is the result of my testing.   Hopefully this will help someone in the future...

[Remote codes for FIOS.xlsx](https://github.com/GerryDazoo/Slinger/files/11240189/Remote.codes.for.FIOS.xlsx)

**Reply by GerryDazoo** [reply]
The code does not support what you're trying to do. Assuming you're trying to remotely restart the slingbox_server executable. You're getting the 404 because you do not have a slingbox named "Restart".

**Reply by GerryDazoo** [reply]
PS. You didn't have to test all those keycodes. When the serve connects it prints out a list of the valid keycodes for the configured remote on that VideoSource.

**Reply by as400jockey** [reply]
It was an interesting chore.  Beats raking leaves.  I presumed that there's nothing above 100...

Prints where?  I'm running the server on Windows... I don't see it spooling anything...

And in this case there seems to be some confusion since MDehrlich also has Fios and yet our "DVR" codes are different.  But everything else lined up OK...

**Reply by as400jockey** [reply]
I just read the release notes for v4 and let me just say, Gerry...

You are the MAN!  Looking forward to playing with SP.
